File: naked-generator/Primary/c/bms_hv.c
#include <string.h>
#include <assert.h>
#include <stdio.h>

#include "bms_hv.h"

void serialize_SET_TS_STATUS(SET_TS_STATUS* set_ts_status, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(SET_TS_STATUS));
	memcpy(buffer, set_ts_status, sizeof(STEER_STATUS));
}

void deserialize_SET_TS_STATUS(uint8_t* buffer, size_t buf_len, SET_TS_STATUS* set_ts_status) {
	assert(buf_len >= sizeof(SET_TS_STATUS));
	memcpy(set_ts_status, buffer, sizeof(STEER_STATUS));
}

File: naked-generator/Primary/py/bms_hv.py
from enum import Enum
from struct import pack, unpack
from collections import namedtuple


class Ts_Status_Set(Enum):
	OFF = 0
	ON = 1


SET_TS_STATUS = namedtuple('SET_TS_STATUS', 'ts_status_set')
SET_TS_STATUS_schema = '<c'


def serialize_SET_TS_STATUS(struct: SET_TS_STATUS) -> bytearray:
	return pack(SET_TS_STATUS, *tuple(struct))


def deserialize_SET_TS_STATUS(buffer: bytearray) -> SET_TS_STATUS:
	return SET_TS_STATUS._make(unpack(SET_TS_STATUS_schema, buffer))


